apps/cf_atom_feed/lib/flags.ex

```diff
@@ -0,0 +1,91 @@
+defmodule CF.AtomFeed.Flags do
+  @moduledoc """
+  Generate an ATOM feed that contains all flags
+  """
+
+  import Ecto.Query
+
+  alias Atomex.{Feed, Entry}
+  alias DB.Schema.{Flag, UserAction, User, Comment}
+  alias DB.Type.FlagReason
+  alias CF.Utils.FrontendRouter
+
+  @nb_items_max 50
+
+  @doc """
+  Get an ATOM feed containing all site's flags in reverse chronological
+  order (newest first)
+  """
+  def feed_all() do
+    flags = fetch_flags()
+    generate_feed(flags, last_update(flags))
+  end
+
+  defp fetch_flags() do
+    DB.Repo.all(
+      from(
+        flag in Flag,
+        order_by: [desc: flag.inserted_at],
+        left_join: action in assoc(flag, :action),
+        left_join: comment in assoc(action, :comment),
+        left_join: statement in assoc(comment, :statement),
```

Comment:
You don't need to load the statement, `action.video_id` already contains the info you need